conans/client/graph/graph_builder.py

```diff
@@ -250,8 +250,18 @@ def _create_new_node(self, node, require, graph, profile_host, profile_build, gr
         new_node.recipe = recipe_status
         new_node.remote = remote
 
-        # FIXME
-        down_options = node.conanfile.up_options
+        # The consumer "up_options" are the options that come from downstream to this node
+        if require.options is not None:
+            # If the consumer has specified requires(options=xxx), we need to use it
+            # It will have less priority than downstream consumers
+            down_options = Options(options_values=require.options)
+            down_options.scope(new_ref.name)
+            # TODO: discuss, together with build_require override, if it is "build" or "visible"
+            if require.visible:  # Only visible requirements propagate options from downstream
```

Comment:
Changed this from ``not require.build`` => ``require.visible``., thinking about what makes more sense...